spec/views/posts_index_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'rails_helper'
+
+RSpec.describe 'posts/index.html.erb', type: :feature do
+  before(:each) do
+    visit new_user_session_path
+    @user = User.new(name: 'Sam', bio: 'Full stack developer', email: 'sam@gmail.com', password: 'sam22021',
+                     post_counter: 1, confirmed_at: Time.now)
+    @user.save
+    (1..5).each do |n|
+      post = Post.create(title: "First post #{n}", text: 'Post content', user_id: @user.id, comments_counter: 0,
+                         likes_counter: 0)
+      post.counter_updater
+    end
+    fill_in 'user[email]', with: 'sam@gmail.com'
+    fill_in 'user[password]', with: 'sam'
+    visit(user_posts_path(@user.id))
+    @post = @user.posts.last
+  end
```

Comment:
- [ ] Most of your tests still failed, which was pointed out by the previous code reviewer. From the failure message on the terminal, the problem comes from when you have something like this;

```ruby
 def new
    like = Like.new
    respond_to do |format|
      format.html { render :new, locals: { like: } }
    end
  end
```

But please modify all the places you have something like the above code snippet to have a different variable name so you can pass it as a key-value pair 👇 to see if your tests pass.

```ruby
 def new
    new_like = Like.new
    respond_to do |format|
      format.html { render :new, locals: { like: new_like } }
    end
  end
```

<img width="517" alt="Code_9PsbxnmvuW" src="https://user-images.githubusercontent.com/81468675/180227833-afe43cb5-0824-4292-8b2c-5668c0286c10.png">

Reply:
Thank you. I saw that comment from the last reviewer, however, whenever I run the specs, I do not see those errors which was why I could not fix it. I have removed the part that was causing those errors from your suggestion.